.github/workflows/_ios-build-test.yml

```diff
@@ -67,27 +67,20 @@ jobs:
           # Install dependencies
           brew install libtool
 
-      - name: Install conda and dependencies
-        run: |
-          # Install conda, setup-miniconda messes with the path that messes with the ruby stuff we do later on
-          curl --retry 3 --retry-all-errors -o "${RUNNER_TEMP}/conda.sh" https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
-          chmod +x "${RUNNER_TEMP}/conda.sh"
-          /bin/bash "${RUNNER_TEMP}/conda.sh" -b -p "${RUNNER_TEMP}/anaconda"
-          echo "${RUNNER_TEMP}/anaconda/bin" >> "${GITHUB_PATH}"
-          # shellcheck disable=SC1091
-          source "${RUNNER_TEMP}/anaconda/bin/activate"
-          conda install -y \
-            blas=1.0 \
-            cffi=1.15.1 \
-            cmake=3.22.1 \
-            mkl=2022.1.0 \
-            mkl-include=2022.1.0 \
-            ninja=1.10.2 \
-            numpy=1.23.3 \
-            pyyaml=6.0 \
-            requests=2.28.1 \
-            setuptools=63.4.1 \
-            typing_extensions=4.3.0
+      - name: Setup miniconda for iOS device
+        if: inputs.ios-platform != 'SIMULATOR'
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
+        with:
+          python-version: "3.10"
+          environment-file: .github/requirements/conda-env-iOS
+
+      - name: Setup miniconda for iOS simulator
```

Comment:
I think that you might not need to do a check here an just install both conda env and pip requirements always.  Some extra pip requirements are fine because they are cached

Reply:
Fixed accordingly. I was afraid that adding requirements could increase the size of the build image.